text/0000-build-std.md

```diff
@@ -0,0 +1,2390 @@
+dependencies:
+
+- `libunwind` will either link to the LLVM `libunwind` or the system's
+  `libunwind`/`libgcc_s`. LLVM's `libunwind` is shipped as part of the
+  rustup component for the standard library and will be linked against
+  when `-Clink-self-contained` is used
+  - This only applies to Linux and Fuchsia targets
+- `compiler_builtins` has an optional `c` feature that will use optimised
+  routines from `compiler-rt` when enabled, which is enabled for the pre-built
+  standard library
+- `compiler_builtins` has an optional `mem` feature that provides symbols
+  for common memory routines (e.g. `memcpy`)
+  - It isn't used by when `std` is built as `libc` provides these routines,
+    but is often used by `no_std` crates when there is not a system `libc`
+- To use sanitizers, the sanitizer runtimes from LLVM's compiler-rt need to
+  be linked against. Building of these is enabled in `bootstrap.toml`
+  ([`build.sanitizers`][bootstrap-sanitizers]) and they are not currently
+  included in the rustup components shipped by the project.
+
+## Cargo features of the standard library
+[cargo-features-of-the-standard-library]: #cargo-features-of-the-standard-library
+
+There are a handful of features defined in the standard library crates'
+`Cargo.toml`s. There is currently no stable existing mechanism for users to
+enable or disable these features. The default set of features is determined by
+[logic in bootstrap][bootstrap-features-logic] and [the `rust.std-features`
+key in `bootstrap.toml`][bootstrap-features-toml]. The enabled features are
+often different depending on the target.
+
+## Target support of the standard library
+[target-support-of-the-standard-library]: #target-support-of-the-standard-library
+
+The `std` crate's `build.rs` checks for supported values of the
+`CARGO_CFG_TARGET_OS` environment variable. `CARGO_CFG_TARGET_OS` corresponds to
+the "os" part of a target triple, e.g. "linux" in "aarch64-unknown-linux-gnu".
+
+When encountering an unknown or unsupported operating system then the
+`restricted_std` cfg is set. `restricted_std` marks the entire standard library
+as unstable, requiring `feature(restricted_std)` to be enabled on any crate
+that depends on it. There is no mechanism for users to enable the `restricted_std`
+feature on behalf of dependencies.
+
+## Panic strategies
+[panic-strategies]: #panic-strategies
+
+There are two panic crates in the standard library - `panic_unwind` and
+`panic_abort` - that provide implementations of panic behaviour. rustc has
+a `-Cpanic` flag that allows the user to chose the panic strategy, though the
+"unwind" option may not be chosen if the target does not support it.
+
+Both crates are compiled and shipped with the pre-built standard library for 
+targets which support `std`. All targets which support the `std` crate default
+to the unwinding panic strategy. Some targets with a pre-built standard library
```

Comment:
There are a number of `std` targets that default to `abort` (like wasi, hermit, espidf, etc.).

Reply:
Fixed, just removed that sentence